docs/user-guide/app_dependencies.md

```diff
@@ -0,0 +1,121 @@
+
+Each selector can match Applications according to one or more properties. If you specify more than one matcher in a selector, each matcher has to match in order for an Application to be considered a dependency. For example, if you specify a `labelSelector` and a `namePattern`, both will have to match in order for an Application to be selected.
+
+Additionally, the behaviour of dependency resolution can be configured on a per Application basis.
+
+## Dependency selectors
+
+Currently, the available properties to match against are `labels` and the `name` of the Application. Each selector can yield zero or more applications.
+
+### Matching on labels
+
+It is possible and recommended to use a standard label selector to select the dependencies for your Application:
+
+```yaml
+spec:
+  dependsOn:
+    selectors:
+    - labelSelector:
+        matchLabels:
+            name: some-app
+```
+
+The above YAML will have the Application depend on any other Application that matches the given label selector. In this case, any application carrying a label `name` with the value of `some-app`.
+
+### Matching on name patterns
+
+Dependencies can be matched on the name of Applications by specifying the `namePattern` selector. The `namePattern` selector takes a list of Application name patterns to match against the Applications' `.metadata.name` property. Specifying multiple patterns 
+
+```yaml
+spec:
+  dependsOn:
+    selectors:
+    - namePattern:
+      - some-app
+      - other-app
+```
+
+The above YAML will have the Application depend on the Applications `some-app` and `other-app`. 
+
+The `namePattern` matcher supports shell-style glob matching, so the following example would match all Applications whose name is starting with `stage1-`:
+
+```yaml
+spec:
+  dependsOn:
+    selectors:
+    - namePattern:
+      - stage1-*
+```
+
+Regular expressions are not supported.
+
+## Configuring behaviour of dependency resolution
+
+There are several options to configure the behaviour of dependency resolution. Please note that these settings are not inherited to dependencies, but must be configured for each dependant uniquely.
+### Delayed start of automated sync
+
+In certain scenarios, you may want to delay the automatic sync process when there are dependencies configured between applications that are created at the same time (i.e. synced through an app of apps pattern from Git). This may allow the synchronization of all applications to succeed before the dependency tree is resolved and the sync is actually started in the right order.
+
+To configure the sync delay, specify the number of seconds to delay in the field `syncDelay` as follows:
+
+```yaml
+spec:
+  dependsOn:
+    syncDelay: 5
```

Comment:
Is delay the only options when is a dependency considered meet, is the requirement just synced what about health checks or  status checks within the resources. I am just thinking out loud an example being I have an operator that creates AWS resources the syncing of that resource does not mean that it's healthy/ready but that my ALB CRD has some health check or `.status.alb.dnsName` being set. This could maybe be documented somewhere?

Reply:
It waits for dependent applications to become healthy. So you may want to create a [custom health check](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#custom-health-checks) for your CRD?

Reply:
Yes, the dependencies feature behave like sync waves, i.e. it inspects the dependencies' sync and health states. 

The `syncDelay` controls how long to wait before checking the status. For example, consider all applications created at the same time, with auto sync turned on. The application controller will start reconciling & syncing whichever app wins the race, so if a dependent app wins the race to a dependency, the dependency's health status might not yet reflect a progressing state. So `syncDelay` gives the application controller a chance to properly inspect dependencies' status. This is also similar to how [sync waves behave](https://github.com/argoproj/argo-cd/blob/129cf5370f9e2c6f99c9a5515099250a7ba42099/controller/sync.go#L534).

Reply:
FTR, I updated the docs to state how dependencies are satisfied